Create session 6.0 and servlet 6.0 projects

pmd1nh opened the pull request:
1. Create Session 6.0 and Servlet 6.0 projects.  
2. Refactor Session in order to reuse in Session 6.0
3. Servlet 6.0 has just the change to make session works with the new jakarta.servlet.http.* APIs
4. Update one existing WC class to replace the javax.servlet.http.HttpUtils which is removed from servlet API (needed for a test)
5. Update SIP
6. Update Event Logging

pmd1nh committed:
Create session 6.0 and servlet 6.0 projects

pmd1nh committed:
Enable tests from issue 20854 for EE10_FEATURES

pmd1nh commented:
#build

pmd1nh commented:
#build

pmd1nh commented:
all passed, including those disabled tests due to session problem.  2 failed tests (may need to check them out..)

com.ibm.ws.request.probe.servlet_fat FAT tests | 2 | 1 | 0 |   |  
-- | -- | -- | -- | -- | --
com.ibm.ws.request.timing_fat FAT tests | 10 | 1 | 0 |   |  

pmd1nh commented:
#build

Full sessions and WC
        #spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_IWAS,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

pmd1nh commented:
Same 2 failed tests show up in FULL build requests. Definitely something is going on

com.ibm.ws.request.probe.servlet_fat FAT tests | 2 | 1 | 0 |   |  
-- | -- | -- | -- | -- | --
com.ibm.ws.request.timing_fat FAT tests | 10 | 1 | 0 |   |  

pmd1nh commented:
All Session buckets pass
--
com.ibm.ws.session_fat FAT tests | 4 | 0 | 0 |   |  
com.ibm.ws.session_fat FAT tests | 61 | 0 | 0 |   |  
com.ibm.ws.session_fat_lWAS FAT tests | 1 | 0 | 0 |   |  
com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration FAT tests | 1 | 0 | 0 |   |  
com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration FAT tests | 1 | 0 | 0 |   |  
com.ibm.ws.session.cache_fat FAT tests | 122 | 0 | 0 |   |  
com.ibm.ws.session.cache_fat FAT tests | 72 | 0 | 0 |   |  
com.ibm.ws.session.cache_fat_config FAT tests | 12 | 0 | 0 |   |  
com.ibm.ws.session.cache_fat_config FAT tests | 26 | 0 | 0 |   |  
com.ibm.ws.session.cache_fat_config_infinispan FAT tests | 12 | 0 | 0 |   |  
com.ibm.ws.session.cache_fat_config_infinispan FAT tests | 5 | 0 | 0 |   |  
com.ibm.ws.session.cache_fat_infinispan FAT tests | 36 | 0 | 0 |   |  
com.ibm.ws.session.cache_fat_infinispan FAT tests | 60 | 0 | 0 |   |  
com.ibm.ws.session.cache_fat_infinispan_container FAT tests | 136 | 0 | 0 |   |  
com.ibm.ws.session.cache_fat_infinispan_container FAT tests | 88 | 0 | 0 |   |  
com.ibm.ws.session.cache_provider_fat FAT tests | 1 | 0 | 0 |   |  
com.ibm.ws.session.cache_provider_fat FAT tests | 1 | 0 | 0 |   |  

pnicolucci commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/SessionContextRegistry.java:
```diff
@@ -141,8 +141,15 @@ public SessionManagerConfig getServerSMC() {
+     * 
+     *  // sd will be null if fromRemote is true
+     * 
+     *  SessionData will be null for the outsider (i.e fromRemote == true). So we can change to AbstractSessionData?
```
Some of these comments should be cleaned up.

pnicolucci commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/SessionData.java:
```diff
@@ -163,8 +63,11 @@ public Object getAttribute(String pName) {
+    
+    /* 
+     * Servlet 6 Update: 
+     *  - Keep here to support getValueNames(); looping of move to AbstractSessionData
```
looping of move?

pmd1nh replied:
looping if move that method into AbstractSessionData

volosied commented on dev/io.openliberty.webcontainer.servlet.6.0.internal/src/io/openliberty/webcontainer60/session/impl/HttpSessionContextImpl60.java:
```diff
@@ -0,0 +1,71 @@
+ * Since Servlet 6.0
+ */
+
+public class HttpSessionContextImpl60 extends HttpSessionContext31Impl {
```
Does  HttpSessionContextImpl60 need to extend HttpSessionContext31Impl? or Can it extend HttpSessionContextImpl  (com.ibm.ws.webcontainer.session.impl.HttpSessionContextImpl) directly instead?

pmd1nh replied:
I will keep this and try out later

jhanders34 commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/AbstractHttpSessionFacade.java:
```diff
@@ -0,0 +1,201 @@
+        _session.readExternal(pIn);
+    }
+
+    // LIBERTY implemented in webcontainer class IHttpSessionFacadeImpl
```
Should these comments be removed?  There is no IHttpSessionFacadeImpl

pmd1nh replied:
removed comments

jhanders34 commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/AbstractHttpSessionFacade.java:
```diff
@@ -0,0 +1,201 @@
+
+public abstract class AbstractHttpSessionFacade implements IBMSession, IBMSessionExt { 
+
+    protected transient AbstractSessionData _session = null;
```
You probably need to have a serialVersionUID for this class as well.

Does this class or any of the serialized sessions need to be deserialized between versions of liberty?  With JCache session persistence, if one app server instance is on the old version and we try to read it in the new version, I am not sure it will work.  It is something that will need to be attempted.  Something you could easily do with a unit test to check in a file that has a HttpSessionFacade and SessionData object serialized with 3.1, 4.0 and 5.0 and make sure each of those versions can read it in.  I am afraid that may not work with the current implementation and will require us to have duplicated code unfortunately.  There is the possibility to get it working and I can work with you on making it right, but we will for sure need to maintain the serialized version uid of the final impl and probably the read and writeExternals should only be on that final subclass and not on the Abstract version of the class.  Regardless it does require some testing.

volosied replied:
@jhanders34  What's liberty's support position here? Should any server level be compatible with another regarding session serialization? No breakage whatsoever? 
- For instance, 17.0.0.4 should still be compatible with 22.0.0.4. 

 Or is it more-so with features and not with server levels? 
 - For instance, all servlet-3.1, or servlet-4.0 should be compatible with each other (regardless of server level). 
 
 @pmd1nh @pnicolucci  Feel free to add any comments in case I'm missing something else.

jhanders34 replied:
If things are stored in a JCache cache or DB and then you upgrade and a session is refreshed from that DB, we should not break serialization compatibility.  It should be an easy test to write to make sure that we haven't broken things.  When things failover to an updated server when doing a rolling upgrade is the main scenario I am concerned with.

jhanders34 replied:
Ok I tried serializing SessionData and its subclasses and HttpSessionFacade and its subclasses.  They cannot be serialized today and http session replication does things differently so this is a non issue.  The reason that it doesn't work today is because there is not no argument constructor.  So it can't be deserialized.

jhanders34 commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/AbstractSessionData.java:
```diff
@@ -0,0 +1,463 @@
+    private final static int SET_IBM_APPLICATION_SESSION = 3;
+
+    private final static String methodNames[] = { "putSessionValue", "getSessionValue", "removeSessionValue", "setIBMApplicationSession" };
+
```
I am surprised there isn't an AbstractSessionData constructor and an abstract AbstractHttpSessionFacade returnFacade() method as well as a protected AbstractHttpSessionFacade _httpSessionFacade field

pmd1nh replied:
added constructor and associated super() from derived class. Moved _httpSessionFacade into common AbstractSessionData and provided a return method

jhanders34 replied:
I was envisioning that:
```
_httpSessionFacade = returnFacade();
```
would still be here and you would have 
```
abstract protected HttpSessionFacade returnFacade();
```
defined on this class that the subclasses would implement.

I would be fine with it being called createFacade actually which makes a lot more sense.

jhanders34 replied:
In reality can't most of the logic in the subclass constructors be consolidated to this constructor so that it mostly just needs to call super in the subclasses?  Remove the dual maintenance and puts the initialization of the fields in the class where they are defined.

pmd1nh replied:
1.  abstract protected HttpSessionFacade returnFacade();
That would cause  subclass HttpSessionImpl to be an abstract class which can break the current flow.
Instead, I made a 
    protected void createFacade(AbstractHttpSessionFacade sessionFacade) 

 and call in from subclasses to set the _httpSessionFacade.
    
2. Consolidated most of the subclasses' constructor logic to AbstractSessionData.  Subclasses constructors need to call into new method createFace(AbstractHttpSessionFacade) to set the session facade.

jhanders34 commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/AbstractSessionData.java:
```diff
@@ -0,0 +1,463 @@
+     * @see com.ibm.wsspi.servlet.session.IBMSessionExt#invalidateAll(boolean)
+     */
+    public void invalidateAll(boolean remote) throws SecurityException {
+        if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && LoggingUtil.SESSION_LOGGER_CORE.isLoggable(Level.FINER)) {
```
Since we are doing a new class, you have the opportunity to change this to com.ibm.websphere.ras to remove use of com.ibm.ejs.ras.

pmd1nh replied:
updated to use websphere.ras...

jhanders34 commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/AbstractSessionData.java:
```diff
@@ -0,0 +1,463 @@
+        super.invalidate();
+    }
+
+    // LIBERTY Moved to webcontainer
```
Maybe can remove this commented out code?

jhanders34 commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/HttpSessionFacade.java:
```diff
@@ -10,105 +10,22 @@
     private static final long serialVersionUID = 3108339284895967670L;
 
-    public HttpSessionFacade(SessionData data) {
+    public HttpSessionFacade(AbstractSessionData data) {
```
Could this stay SessionData?  That is the AbstractSessionData that goes with this class right?

jhanders34 commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/SessionContext.java:
```diff
@@ -424,7 +424,7 @@ public int getReaperInterval(long timeout) {
     }
 
-    protected void setSIPCookieIfApplicable(HttpServletRequest _request, HttpServletResponse response, SessionData sd) {
+    protected void setSIPCookieIfApplicable(HttpServletRequest _request, HttpServletResponse response, AbstractSessionData sd) {
```
Can these stay SessionData.  It won't be any other implementation besides that one right?

jhanders34 commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/SessionData.java:
```diff
@@ -10,151 +10,51 @@
-    protected HashMap locks = null;
-
-    protected String pathInfoForAppSession = null;
     String sipCookieInfo = null;
```
sipCookieInfo is in the abstract class.  No need to duplicate it here.

pmd1nh replied:
cleanup

jhanders34 commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/SessionData.java:
```diff
@@ -10,151 +10,51 @@
+    private static final String methodClassName = "SessionData";
 
-    protected boolean _hasSecurityInfo = true; // used in
+    //protected boolean _hasSecurityInfo = true; // used in
```
Should remove this commented out code.

jhanders34 commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/AbstractSessionData.java:
```diff
@@ -0,0 +1,463 @@
+    /*
+     * @see javax.servlet.http.HttpSession#getValueNames()
+     */
+    public String[] getValueNames() {
```
I am surprised you are leaving getValue, getValueNames, putValue, and removeValue on the abstract version since it is removed in 6.0.

pmd1nh replied:
Clean up those removed methods

jhanders34 commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/http/AbstractHttpSession.java:
```diff
@@ -0,0 +1,239 @@
+     * Adding a message for when IllegalStateException is thrown
+     */
+    private static final String iseMessage = "The method is called on an invalidated session: ";
+
```
Again I would suggest an AbstractHttpSession constructor that the subclasses can delegate to.

pmd1nh replied:
added constructor for abstract classes and called super() from its derived classes

jhanders34 commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/http/HttpSessionImpl.java:
```diff
@@ -14,64 +14,42 @@
+ * Common APIs are moved up into the AbstractHttpSession
  */
-public class HttpSessionImpl implements HttpSession {
+public class HttpSessionImpl extends AbstractSessionData {
```
This change completely modifies the behavior of the HttpSession methods.  AbstractSessionData's logic is different than AbstractHttpSession's logic.  This needs to be changed to not change behavior.

zhang-ca replied:
I see HttpSessionImpl extends AbstractSessionData extends AbstractHttpSession implements HttpSession
which should work fine with old servlet 3.1.

pmd1nh replied:
@jhanders34 can I request a second look on this hierarchy change for 

public class HttpSessionImpl extends AbstractSessionData

from the surface, it looks like a behavior change.  However, the methods in AbstractSessionData do some extra checks:

1. crossoveCheck() - this is for debugging purpose and requires additional configuration to enable.
2. Check session name against "com_ibm_ejs_security_httpsession_info" and securityInfo - I cannot find where/whom may use/set this session name.  @zhang-ca thinks it is a left over from tWAS migration to OL.

then delegating to parent AbstractHttpSession's methods.

I don't see real behavior differences from the new hierarchy.

zhang-ca replied:
"com_ibm_ejs_security_httpsession_info" is for Admin Console on tWAS. It wouldn't be used in Liberty.

jhanders34 replied:
I am going to mark this resolved, but I am going to look at cleaning this up after this PR is merged.

jhanders34 commented on dev/io.openliberty.session.6.0.internal/src/io/openliberty/session/SessionData60.java:
```diff
@@ -0,0 +1,116 @@
+    /*
+     * @see javax.servlet.http.HttpSession#getValue(java.lang.String)
+     */
+    @Override
```
Why are getValue, putValue and removeValue on this implementation?  Should getValueNames be made private?

jhanders34 commented on dev/io.openliberty.webcontainer.servlet.6.0.internal/resources/com/ibm/ws/webcontainer60/resources/Messages.nlsprops:
```diff
@@ -0,0 +1,53 @@
+#
+# This file has been modified to follow the WebSphere Message Guidelines.
+# For more information, visit:
+#   http://washome.austin.ibm.com/xwiki/bin/view/InformationDevelopmentTeam/MESSAGES_DISTRIBUTED
```
A messages file with everything commented out?  Very odd.

pmd1nh replied:
we will probably have some new servlet 6 messages so just thinking to have the file in place.  I can remove until we really need one.

pmd1nh replied:
deleted to avoid any confusion.

jhanders34 commented on dev/io.openliberty.webcontainer.servlet.6.0.internal/src/io/openliberty/webcontainer60/session/impl/HttpSessionContextImpl60.java:
```diff
@@ -0,0 +1,71 @@
+    public HttpSessionContextImpl60(SessionManagerConfig smc, SessionApplicationParameters sap, SessionStoreService sessionStoreService) {
+        super(smc, sap, sessionStoreService);
+        
+        if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled()&&LoggingUtil.SESSION_LOGGER_CORE.isLoggable(Level.FINE)) {
```
similar to earlier comment, you should switch to com.ibm.websphere.ras in a new class.

pmd1nh replied:
changed to com.ibm.websphere.ras.TraceComponent

jhanders34 commented on dev/io.openliberty.webcontainer.servlet.6.0.internal/src/io/openliberty/webcontainer60/session/impl/HttpSessionContextImpl60.java:
```diff
@@ -0,0 +1,71 @@
+    /**
+     * create chain wc.HttpSesionImpl60 > SessionData60 > ses.HttpSessionImpl60
+     */
+    public Object createSessionObject(ISession isess, ServletContext servCtx) {
```
Should these methods have the `@Override` annotation?

pmd1nh replied:
added

jhanders34 replied:
`these` meant updating generateNewId as well to have `@Override`

jhanders34 commented on dev/io.openliberty.webcontainer.servlet.6.0.internal/src/io/openliberty/webcontainer60/session/impl/HttpSessionImpl60.java:
```diff
@@ -0,0 +1,73 @@
+    }
+
+    /*
+     * For security to store away special hidden value in the session
```
How are the methods in this class used?  I don't see them in the current implemenation.

pmd1nh replied:
I can't find where the webcontainer.session.impl.HttpSessionFacadeImpl is instantiated.  It is the only class which calls into get and putSecurityInfo().

zhang-ca replied:
I don't think security hidden value is being used in Liberty if no one called [com.ibm.ws](http://com.ibm.ws/).webcontainer.session.impl.HttpSessionImpl.getSecurityInfo()

zhang-ca replied:
com.ibm.ws.webcontainer.session.impl.HttpSessionFacadeImpl.getSecurityInfo()
com.ibm.ws.webcontainer.session.impl.HttpSessionFacadeImpl.putSecurityInfo(Object)

Based on the WAS change history, the 2 methods were added by Admin Console. Most likely not being used in Liberty

jhanders34 replied:
Seems like the HttpSessionFacadeImpl class can be deleted. Also getSecurityInfo and putSecurityInfo methods can be removed from multiple places then as well.

jhanders34 replied:
As stated above, going to mark resolved, but will clean up in a PR after this PR is merged.

jhanders34 commented on dev/io.openliberty.webcontainer.servlet.6.0.internal/src/io/openliberty/webcontainer60/session/impl/SessionContextRegistryImpl60.java:
```diff
@@ -0,0 +1,54 @@
+    /*
+     * (non-Javadoc)
+     *
+     * @see com.ibm.ws.webcontainer.session.impl.SessionContextRegistryImpl#addHttpSessionIdListeners(java.util.ArrayList, java.lang.String)
```
Should remove commented out code.

pmd1nh replied:
removed

jhanders34 commented on dev/io.openliberty.webcontainer.servlet.6.0.internal/src/io/openliberty/webcontainer60/session/impl/HttpSessionImpl60.java:
```diff
@@ -0,0 +1,73 @@
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.webcontainer60.session.impl;
```
If following the new naming conventions for liberty, each of these packages should have internal added to them, so for this one it would be io.openliberty.webcontainer60.internal.session.impl

pmd1nh replied:
updated packages for webcontainer60 and session 60

pmd1nh commented:
#build

Full sessions, JSF, JSP and WC
#spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_IWAS,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.3_fat, com.ibm.ws.jsf_fat,com.ibm.ws.jsf_fat_lWAS

pmd1nh commented:
resubmit, previous build exited after 3hrs

#build

Full sessions, JSF, JSP and WC

#spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_IWAS,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.3_fat, com.ibm.ws.jsf_fat,com.ibm.ws.jsf_fat_lWAS

jhanders34 commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/SessionContext.java:
```diff
@@ -439,12 +439,14 @@ protected void setSIPCookieIfApplicable(HttpServletRequest _request, HttpServlet
             LoggingUtil.SESSION_LOGGER_CORE.entering(methodClassName, methodNames[SESSION_POST_INVOKE]);
         }
-        SessionData s = (SessionData) sess;
+        AbstractSessionData s = (AbstractSessionData) sess;
```
Can this stay SessionData like was done with setSIPCookieIfApplicable?

jhanders34 commented on dev/com.ibm.ws.session/src/com/ibm/ws/session/SessionData.java:
```diff
@@ -10,151 +10,40 @@
-
-    protected final static WebSphereRuntimePermission invalidateAllPerm = new WebSphereRuntimePermission("accessInvalidateAll");
-
+    //protected HttpSessionFacade _httpSessionFacade;
```
Shouldn't have this commented out code.

jhanders34 commented on dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/session/impl/HttpSessionImpl.java:
```diff
@@ -56,7 +56,7 @@ public Object getSecurityInfo()
   public Object getFacade()
   {
-    return (Object) _httpSessionFacade;
+    return (Object) returnFacade();
```
Why this behavior change?  returnFacade news up a new instance.

pmd1nh replied:
undo

jhanders34 commented on dev/io.openliberty.session.6.0.internal/src/io/openliberty/session/internal/http/HttpSessionImpl60.java:
```diff
@@ -0,0 +1,54 @@
+    protected HttpSessionImpl60(ISession session) {
+        super(session);
+
+        if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && LoggingUtil.SESSION_LOGGER_CORE.isLoggable(Level.FINE)) {
```
Another com.ibm.ejs.ras instead of com.ibm.websphere.ras

jhanders34 commented on dev/io.openliberty.webcontainer.servlet.6.0.internal/bnd.bnd:
```diff
@@ -0,0 +1,64 @@
+# Append ";provide:=true" if this bundle also provides an implementation
+# for the exported API.
+Export-Package: \
+    io.openliberty.webcontainer60.internal.session.impl*;provide:=true
```
If it is exported it shouldn't have `internal` in the package name.

jhanders34 replied:
Is it possible to combine this bundle with the factories bundle?

jhanders34 replied:
@tevans78 can correct me though.  `internal` should only be used for packages that are not exported from a bundle right?

pnicolucci replied:
Regarding combining we don't want to do that. The factories bundle is where `com.ibm.ws.webcontainer.speclevel` exists. There is a properties file that allows us to determine which version of the Servlet specification is in use in our common webcontainer code. Only one factories bundle is loaded per server feature. However, the non factories servlet spec specific bundles (serv.et.3.1, servlet.4.0, servlet.6.0) can be loaded by more than one feature.

jhanders34 replied:
Exported packages still have internal in them.  Can fix this in a follow on PR since it is servlet 6.0 specific code.  This makes things messy for io.openliberty.servlet.6.0.internal.factories/bnd.bnd leaving it with internal in the package name.

pmd1nh replied:
Removed .internal. in 2 packages: io.openliberty.webcontainer60.internal.session.impl -> io.openliberty.webcontainer60.session.impl and io.openliberty.session.internal - > io.openliberty.session.impl

jhanders34 replied:
Not sure if both were needed to be done.  Were both imported by other bundles?  I will leave it as it is right now and move forward.

pmd1nh commented:
#build

Full sessions, JSF, JSP and WC

#spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_IWAS,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.3_fat, com.ibm.ws.jsf_fat,com.ibm.ws.jsf_fat_lWAS

pmd1nh commented:
#build

#spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_IWAS,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.3_fat, com.ibm.ws.jsf_fat,com.ibm.ws.jsf_fat_lWAS

pmd1nh committed:
Update Event Logging tests for session

zhang-ca reviewed (approved):
All session tests on Hazelcast and Infinispan cache pass with repeat EE10 and EE9 runtime.

pmd1nh commented:
Updated event logging FAT to log event "websphere.session.getAttribute" and "websphere.session.setAttribute".  This should address break in "com.ibm.ws.request.probe.servlet_fat FAT tests"

#build

#spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_IWAS,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.3_fat, com.ibm.ws.jsf_fat

pmd1nh commented:
Updated to address a break in SIP test: (FetchFritzResultsForMoon | suiteCONVERGED)

1. Bring back method " AbstractHttpSessionFacade returnFacade() " in SessionData and SessionData60.  ConvergedHttpSessionImpl is a subclass of SessionData and has its own returnFacade() to create WsHttpSessionFacade() which is used for SIP. 

2. Keep AbstractSessionData.createFacade() to populate _httpSessionFacade in one place.

SIP test passes locally with this change.

pmd1nh commented:
#build

#spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_IWAS,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.3_fat, com.ibm.ws.jsf_fat

pmd1nh commented:
a clean PB.  Only one failed but it was an invalid bucket



no component | 89118 | 1 | 0 |   |  
-- | -- | -- | -- | -- | --
Invalid test bucket - com.ibm.ws.session_fat_IWAS | 1 | 1 | 0 |  

pmd1nh commented:
@pgunapal can you review the change for Event Logging?  Thanks!

pmd1nh commented:
fixed a very minor trace spec.

pgunapal reviewed (approved):
The Request Probe changes look good!

pmd1nh commented:
add abstract AbstractHttpSessionFacade returnFacade(); to AbstractSessionData
moved SessionData's returnFacade() up to its parent HttpSessionImpl
moved SessionData60's returnFacade() to its parent HttpSessionImpl60

#build

#spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.3_fat, com.ibm.ws.jsf_fat

pmd1nh commented:
minor cleanup in traces and unused code

pmd1nh commented:
Updated copyright

pmd1nh commented:
com.ibm.ws.collective.repository_fat FAT tests - most of the above failed tests are either in java.net.ConnectException or Connection refused.  Nothing in any of of the session related tests.

resubmit

#build

#spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.3_fat, com.ibm.ws.jsf_fat

pmd1nh commented:
Cancelled the above build  https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_6XxCkOjMEeyVkLDPEXw2dw

Jared shows a better way to re-run just the failed tests.  Thank you!



com.ibm.ws.collective.repository_fat FAT tests | 63 | 0 | 25 |   |  
-- | -- | -- | -- | -- | --
com.ibm.ws.collective.repository_fat FAT tests | 63 | 0 | 0 |   |  
 

pmd1nh commented:
Updated session and webcontainer to remove .internal in package names.

#build
#spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.3_fat, com.ibm.ws.jsf_fat

pmd1nh commented:
3 failed tests in due to Connection refused
Resubmitted the child build for the failed bucket.  3 failed tests are passed now.

com.ibm.ws.jaxws.2.2.webcontainer_fat FAT tests | 110 | 0 | 3 |   |  
-- | -- | -- | -- | -- | --
com.ibm.ws.jaxws.2.2.webcontainer_fat FAT tests | 110 | 0 | 0 |  

pmd1nh commented:
Last rebase before merging

#libby
